### PR TITLE
[WIPTEST] Cloud tenants CRUD 

### PR DESCRIPTION
**Pull request by rrasouli**
Purpose or Intent
=================

A draft 
Adding openstack cloud provider tenant CRUD functions.
Ignore the commit message 'fixed refresh function'

### CODE REVIEW

**Comment by mshriver** [comment]
@rrasouli Please check out #3942, I didn't see your PR and picked up a JIRA card for adding these destinations. Your implementation is good, but is missing some things like paginator handling when there are >20 tenants, or the multiple delete/edit paths (from All, from Details).  Don't want to throw out all of either of our work - its definitely my fault for duplicating your effort though, sorry about that.

Let me know what you think about my changes, and thoughts on how to proceed. Maybe you can focus your efforts on growing the test_tenant test module and I can cover the navigation and object modeling?

**Comment by rrasouli** [comment]
@mshriver I can see your PR and tried to run it, I wonder if we can merge both PR's
I would have split each CRUD test. update tenant is missing, as well as some navigation utilities

**Comment by mshriver** [comment]
@rrasouli I saw your comments on my PR that got merged - I guess it wasn't quite ready but we can resolve that quickly. Do you want to rebase against master and improve the CRUD test and I'll tackle the exception I caused by not setting result when `cancel=True`?

**Comment by rrasouli** [comment]
@mshriver I could not rebase my code here, perhaps we can merge the code in #3942

File: cfme/cloud/tenant.py [hunk 1]
```diff
@@ -7,27 +7,61 @@
...
+        """
+            create a new tenant
+        """
+        navigate_to(Tenant, 'All')
+        tb.select('Configuration', 'Create Cloud Tenant')
+        tenant_data = dict(tenant_name=self.name, provider_name=provider_key)
+        create_form.fill(tenant_data)
+        sel.click(form_buttons.save)
+
+    def update(self, tenant_name, new_name):
```
Comment (mfalesni) [inline cfme/cloud/tenant.py]: Update has a special syntax.

```python
def create(self, updates):
    ...
    ...
    fill(create_form, updates)
    flash.assert_message('tenant {} was saved'.format(updates.get('name', self.name))
```

it is not exact with strings etc but it shows the usage

File: cfme/tests/openstack/cloud/test_tenant_crud.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    new_tenant.create(cloud_provider)
+    wait_for(provider.is_refreshed, [None, 10], delay=5)
+    sel.refresh()
+    assert new_tenant.exists(), "New cloud tenant hasn't been created"
+
+
+@pytest.mark.usefixtures("setup_provider_modscope")
+def test_update_cloud_tenant(cloud_tenant_name, new_tenant_name, cloud_provider):
+    my_tenant = Tenant(cloud_tenant_name, "", cloud_provider)
+    my_tenant.update(cloud_provider, new_tenant_name)
```
Comment (mfalesni) [inline cfme/tests/openstack/cloud/test_tenant_crud.py]:
```python
from utils.update import update

with update(my_tenant):
    tenant.name = 'foo'
    tenant.whatever = 'something'
```

By the way, why not having a complete crud test in one? Like we do it?

File: cfme/cloud/tenant.py [hunk 3]
```diff
@@ -7,27 +7,61 @@
...
         self.description = description
         self.provider_key = provider_key
+        Navigatable.__init__(self)
+
+    def create(self, provider_key):
+        """
+            create a new tenant
+        """
+        navigate_to(Tenant, 'All')
+        tb.select('Configuration', 'Create Cloud Tenant')
```
Comment (mfalesni) [inline cfme/cloud/tenant.py]: You should use an Add location instead of All->click table

File: cfme/cloud/tenant.py [hunk 4]
```diff
@@ -7,27 +7,61 @@
...
     locators={
         'tenant_table': SplitTable(header_data=('//div[@class="xhdr"]/table/tbody', 1),
-            body_data=('//div[@class="objbox"]/table/tbody', 1))
+                                   body_data=('//div[@class="objbox"]/table/tbody', 1))
     },
     title='Cloud Tenants')
 
-
+create_form = Form([('cloud_provider', ("//button[@id='cloud_tenant_id']"),
+                     ('tenant_name', "//*[@name='name']"))])
```
Comment (mfalesni) [inline cfme/cloud/tenant.py]: we have an Input class in cfme.web_ui

File: cfme/tests/openstack/cloud/test_tenant_crud.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+import cfme.fixtures.pytest_selenium as sel
+from cfme.cloud.tenant import Tenant
+from utils.wait import wait_for
+
+
+@pytest.mark.usefixtures("setup_provider_modscope")
+def test_create_cloud_tenant(provider):
+    """
+        create a new cloud tenant
+    """
```
Comment (mfalesni) [inline cfme/tests/openstack/cloud/test_tenant_crud.py]: Can you follow some reasonable test descriptions?
Like this one https://github.com/ManageIQ/integration_tests/blob/master/cfme/tests/test_ui_problems.py#L26

File: cfme/cloud/tenant.py [hunk 6]
```diff
@@ -7,27 +7,61 @@
...
         """Base class for a Tenant"""
         self.name = name
         self.description = description
         self.provider_key = provider_key
+        Navigatable.__init__(self)
+
+    def create(self, provider_key):
+        """
+            create a new tenant
+        """
```
Comment (mfalesni) [inline cfme/cloud/tenant.py]: Comments should look like this: https://google.github.io/styleguide/pyguide.html?showone=Comments#Comments

File: cfme/cloud/tenant.py [hunk 7]
```diff
@@ -38,3 +72,36 @@ def exists(self):
...
+    def resetter(self):
+        tb.select('List View')
+
+
+@navigator.register(Tenant, 'Details')
+class Details(CFMENavigateStep):
+    prerequisite = NavigateToSibling('All')
+
+    def step(self):
+        tb.select('List View')
```
Comment (mfalesni) [inline cfme/cloud/tenant.py]: This one should be as a resetter, shouldnt it?

**Comment by rrasouli** [comment]
Will continue from #3942

**Pull request closed without merging**